pkg/controller/external.go

```diff
@@ -162,25 +162,26 @@ func (e *external) Observe(ctx context.Context, mg xpresource.Managed) (managed.
 		return managed.ExternalObservation{}, errors.Wrap(err, "cannot get connection details")
 	}
 
+	lateInitedParams, err := tr.LateInitialize(res.State.GetAttributes())
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, "cannot late initialize parameters")
+	}
+	lateInitialized := annotationsUpdated || lateInitedParams
+	resourceReady := tr.GetCondition(xpv1.TypeReady).Equal(xpv1.Available())
 	// We try to mark the resource ready before the spec late initialization
```

Comment:
nit: looks like this comment needs to be updated, i.e. we no longer mark as ready before late init

Reply:
If I'm not mistaken, we still prioritize marking as ready over late-initialization, as `ExternalObservation.ResourceLateInitialized` will not be set to `true` if the resource does not have the `Ready` condition. And we prioritize a status update for initially marking the resource as ready over a spec update for actually late-initializing the resource. One of the goals of this change was to preserve that optimization as it affects the [time-to-readiness](https://github.com/crossplane/terrajet/issues/55#issuecomment-929494212) metric we defined for managed resources.